yroots/subdivision.py

```diff
@@ -819,13 +812,13 @@ def subdivision_solve_nd(funcs , a, b, deg, target_deg, interval_data,
     # Used if subdividing further.
     # Only choose good_degs if the approximation after trim_coeffs is good.
     if good_approx:
-        # good_degs are assumed to be 1 higher than the current approx for more
-        # accurate performance.
-        good_degs = [coeff.shape[0] for coeff in coeffs]
+        # good_degs are assumed to be 1 higher than the current approximation
+        # but no larger than the initial degree for more accurate performance.
+        good_degs = [min(coeff.shape[0], deg) for coeff in coeffs]
         good_zeros_tol = max(tols.min_good_zeros_tol, sum(np.abs(approx_errors))*tols.good_zeros_factor)
 
     # Check if the degree is small enough or if trim_coeffs introduced too much error
-    if np.any(np.array([coeff.shape[0] for coeff in coeffs]) > target_deg) or not good_approx:
+    if np.any(np.array([coeff.shape[0] - 1 for coeff in coeffs]) > target_deg) or not good_approx:
```

Comment:
Wouldn't it be slightly faster to do 
np.any(np.array([coeff.shape[0] for coeff in coeffs]) > target_deg +1)

Reply:
That's a good point. I think that' you're right!